### PR TITLE
#508: Optimize event for mezon-server

### PR DESCRIPTION
**Pull request by hieudinhngoc-ncc**
# Ticket
#508 

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: transaction/transaction.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ const (
 	TransactionExtraInfoDonationCampaignFeed = "donation-campaign-feed"
 )
 
+const (
```
Comment (hoangdohuy) [inline transaction/transaction.go]: fix thẳng vào constant cũ

File: network/server_grpc.go [hunk 2]
```diff
@@ -357,6 +358,15 @@ func (s *server) convertEventToStatusUpdate(event events.BlockchainEvent, txHash
 	return nil
 }
 
+func convertExtraInfoToEvent(extraInfo string) string {
+	switch {
+	case strings.Contains(extraInfo, transaction.TransactionExtraInfoGiveCoffee):
```
Comment (hoangdohuy) [inline network/server_grpc.go]: convert ra dict[string][string] -> check ["type"]

**Review by hoangdohuy — APPROVED** [review]
LGTM
